### ISSUE TITLE
Allow external processes to access kafka from the containers

### ISSUE THREAD

**Issue by sfehlandt** [IntegratedAlarmSystem-Group/integration-tools#9]
Currently, external processes cannot to consume-from or produce-to kafka.

They should be able to do so

**Comment by sfehlandt** [comment]
We are going to address this by changing the KAFKA_SERVER environment variable, defined in the .env files.

If instead of the value "kafka" we provide the IP of the host system, external processes will be able to connect using that IP (possibly even localhost if they are running in the same host machine).

This will require the host machine to NOT run kafka nor zookeeper. Also, this could not work depending on the firewall configuration (for example not allowing to open port 9092). Talk to you sys-admin in such a case.

**Comment by acaproni** [comment]
Can you please provide an example where kafka is running in a docker container on a given server (with IP1) and we want to run a plugin in another server (with IP2)?

We need this fixed soon (i.e. before release 4) because without we cannot push new alarms in the IAs including the missing ones for the weather

**Comment by acaproni** [comment]
Another note. It may not be today but atthe leatest for release 4, IAS tools like supervisors and converters must take the URL to connect to kafka from the CDB. We will be able to override through properties as it is useful for testing for example but it will not be the standard way.

**Comment by sfehlandt** [comment]
We can check if it works reading the addresses form the CDB. We can create another docker-compose for that if you want. (I do not see a reason this should not work if you already implemented this in the core).

But for development is better the other way because in that way deployment is independent of where we are actually deploying. So the docker-compose files for development will continue to override the definition of the CDB.

**Comment by sfehlandt** [comment]
> Can you please provide an example where kafka is running in a docker container on a given server (with IP1) and we want to run a plugin in another server (with IP2)?


You can just use the same docker-compose file in the 2 servers. Edit the corresponding .env file so KAFKA_SERVERS property is set to IP1 (The same address IP1 in both files). Then run everything except alma-weather-plugin in the server1 (IP1) and in the other one just run alma-weather-plugin:

SERVER 1 (IP1):
docker-compose -f docker-compose-dev up -d kafka converter supervisor dummy-plugin dummy-supervisor sender webserver display nginx

SERVER 2 (IP2):
docker-compose -f docker-compose-dev up -d alma-weather-plugin

> We` need this fixed soon (i.e. before release 4) because without we cannot push new alarms in the IAs including the missing ones for the weather

I am sorry, I do not understand that comment

**Comment by acaproni** [comment]
Thank you Sebastian.
Actually we have dockers only in the alarm server not in the other one. I will redeploy and check if it works.

To clarify my last comment. We have plugins running in servers other then ias-server01 that must be able to push monitor points in the BSDB. If kafka is not visible from such server then the plugins do not work. One of the plugins that need to be able to connect to kafka from a remote server is the one the get the locations of the antennas in the pads that runs in ape-gns. Another one is the one that gets HVAC and fire alarms.

**Comment by acaproni** [comment]
It does not work from ape1-gns:
```
11-45-00.760 |WARN | [NetworkClient.java:615] [kafka-producer-network-thread | APE1-ANTPAD] [Producer clientId=APE1-ANTPAD] Connection to node -1 could not be established. Broker may not be available.
```

Could it be a problem setting/exporting the IP?
```
[ias@v-iasserv01 ~]$ docker container ls -a
...
0d65751bfd71        alarmsystem/ias-kafka:latest                      "./entrypoint.sh"        3 minutes ago       Up 3 minutes               0.0.0.0:2181->2181/tcp, 0.0.0.0:9092->9092/tcp   ias-kafka
```

Same error running in v-iasserv01, where docker containers run:
```
[ias@v-iasserv01 src]$ iasValueDumper -k v-iasserv01.osf.alma.cl:9092
Traceback (most recent call last):
  File "./../bin/iasValueDumper", line 97, in <module>
...
 File "/opt/anaconda3/lib/python3.5/site-packages/kafka/client_async.py", line 839, in check_version
    raise Errors.NoBrokersAvailable()
kafka.errors.NoBrokersAvailable: NoBrokersAvailable
```

**Comment by sfehlandt** [comment]
Problem was due to other causes.
Closed and improved by: e82bf2ff7a3bab46174239639af00acbb97726a7
